tests/clients/test_report.py

```diff
@@ -0,0 +1,224 @@
+# Copyright 2018 ASI Data Science
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import uuid
+import datetime
+
+import pytest
+from marshmallow import ValidationError
+
+from sherlockml.clients.report import (
+    Report,
+    ReportSchema,
+    VersionedReport,
+    VersionedReportSchema,
+    ReportVersion,
+    ReportClient,
+)
+
+from tests.clients.fixtures import PROFILE
+
+
+PROJECT_ID = uuid.uuid4()
+USER_ID = uuid.uuid4()
+REPORT_ID = uuid.uuid4()
+VERSION_ID = uuid.uuid4()
+PROJECT_ID = uuid.uuid4()
+
+ACTIVE_VERSION = ReportVersion(
+    created_at=datetime.datetime(2018, 10, 3, 9, 23, 5, 0),
+    author_id=USER_ID,
+    report_path="/.sml/tavern/{}/{}/index.html".format(REPORT_ID, VERSION_ID),
+    report_key="{}/.sml/tavern/{}/{}/index.html".format(
+        PROJECT_ID, REPORT_ID, VERSION_ID
+    ),
+    report_bucket="sml-projects-test-bucket",
+    notebook_path="/test-notebook-path.ipynb",
+    report_id=REPORT_ID,
+    id=VERSION_ID,
+)
+
+REPORT = Report(
+    id=REPORT_ID,
+    name="Test Report Name",
+    description="Looking forward to the test reports on this Test Report",
+    created_at=datetime.datetime(2018, 10, 3, 9, 23, 0, 0),
+    active_version=ACTIVE_VERSION,
+)
+
+VERSIONED_REPORT = VersionedReport(
+    id=REPORT_ID,
+    name="Test Report Name",
+    description="Looking forward to the test reports on this Test Report",
+    created_at=datetime.datetime(2018, 10, 3, 9, 23, 0, 0),
+    active_version_id=VERSION_ID,
+    versions=[ACTIVE_VERSION],
+)
+
+
+REPORT_BODY = {
+    "created_at": REPORT.created_at.strftime("%Y-%m-%dT%H:%M:%S.%f")[:-3],
+    "report_name": REPORT.name,
+    "report_id": str(REPORT.id),
+    "description": REPORT.description,
+    "active_version": {
```

Comment:
I would factor this inner dict out from here and from `VERSIONED_REPORT_BODY` as e.g. REPORT_VERSION_BODY` to avoid repetition, and also add a separate `test_report_version_schema` where you explicitly test is deserialisation.